ert/ensemble_evaluator/builder/_legacy.py

```diff
@@ -100,132 +110,159 @@ def _evaluate(self) -> None:
         if self._config is None:
             raise ValueError("no config")
 
-        async def _evaluate_inner() -> None:
-            """
-            This (inner) coroutine does the actual work. It prepares and
-            executes the necessary bookkeeping, prepares and executes
-            the JobQueue, and dispatches pertinent events.
-
-            Before returning, it always dispatches a CloudEvent describing
-            the final result of executing all its jobs through a JobQueue.
-            """
-            # Set up the timeout-mechanism
-            timeout_queue: "asyncio.Queue[CloudEvent]" = asyncio.Queue()
-            on_timeout, send_timeout_future = self.setup_timeout_callback(timeout_queue)
-
-            if not self._ee_id:
-                raise ValueError(f"invalid ensemble evaluator id: {self._ee_id}")
-            if not self._config:
-                raise ValueError("no config")  # mypy
-
-            # event for normal evaluation
-            result = CloudEvent(
+        # The cloudevent_unary_send only accepts a cloud event, but in order to
+        # send cloud events over the network, we need token, URI and cert. These are
+        # not known until evaluate() is called and _config is set. So in a hacky
+        # fashion, we create the partialmethod (bound partial) here, after evaluate().
+        # Note that this is the "sync" version of evaluate(), and that the "async"
+        # version uses a different cloudevent_unary_send.
+        self.__class__._ce_unary_send = partialmethod(  # type: ignore
```

Comment:
Is this partial created just that we don't need to tyhpe token, cert and uri all the time or is it something else too?